benchmarks/monitoring_log_benchmark.ipynb

```diff
@@ -0,0 +1,227 @@
+    "        \"\"\"\"\n",
+    "        [Tue Jul 17 04:14:59 UTC 2018]\n",
+    "        * CPU usage: 8.1%\n",
+    "        * Memory usage: 6%\n",
+    "        * Disk usage: 57%\n",
+    "        \"\"\"\n",
+    "        for line in lines:\n",
+    "            if 'CPU usage:' in line:\n",
+    "                # if it's a cpu line parse to float and add to cpu collector\n",
+    "                cpu = cpu + [float(re.sub(\"[^$0-9.]\",\"\", line))]\n",
+    "            elif 'Memory usage:' in line:\n",
+    "                # if it's a mem line parse to float and add to mem collector\n",
+    "                mem = mem + [float(re.sub(\"[^$0-9.]\",\"\", line ))]\n",
+    "            elif 'Disk usage:' in line:\n",
+    "                # if it's a disk line parse to float and add to disk collector\n",
+    "                disk = disk + [float(re.sub(\"[^$0-9.]\",\"\", line))]\n",
+    "            elif '[' in line:\n",
+    "                # if it's a time line parse to datetime object and add to time collector\n",
+    "                time = time + [parser.parse(line.replace(\"[\", '').replace(\"]\", '').strip())]\n",
+    "        # create a dataframe to make adding to plotly easier\n",
+    "        df = pd.DataFrame(data={'cores': cpu, 'memory usage (G)': mem, 'disk usage (G)': disk, 'time':time})\n",
+    "        # create the title- eg \"Usage: 64.0 Cores, 236.0 G Memory, 394.0 G Disk Space\" \n",
+    "        title = \"Usage: \" + str(cpu_cap) + \" Cores, \" + str(mem_cap) + \" G Memory, \" + str(disk_cap) + \" G Disk Space\"  \n",
+    "        # CPU usage trace (Note the legend group)\n",
+    "        trace1 = go.Scatter(\n",
+    "            x = df.time,\n",
+    "            y = df['cores'],\n",
+    "            name = monitor + ' core usage (G)',\n",
+    "            legendgroup =  'core usage (G)',\n",
+    "        )\n",
+    "\n",
+    "        # Memory Usage trace (Note the legend group)\n",
+    "        trace2 = go.Scatter(\n",
+    "            x = df.time,\n",
+    "            y = df['memory usage (G)'],\n",
+    "            name = monitor + ' memory usage (G)',\n",
+    "            legendgroup =  'memory usage (G)',\n",
+    "        )\n",
+    "\n",
+    "        # Disk Usage trace (Note the legend group)\n",
+    "        trace3 = go.Scatter(\n",
+    "            x = df.time,\n",
+    "            y = df['disk usage (G)'],\n",
+    "            name = monitor + ' disk usage (G)',\n",
+    "            legendgroup =  'disk usage (G)',\n",
+    "        )\n",
+    "        \n",
+    "        # Add to data collector for plotly\n",
+    "        data = data + [trace1,trace2,trace3]\n",
+    "        \n",
+    "        # get the earliest time (aka start of task) and add it to early collector\n",
+    "        earlys = earlys + [min(time)]\n",
+    "    # TODO figure out a better way to make sure these annotations, which can be long, do not overlap\n",
+    "    def height(num, x = 40, maxh = 60, by = 10):\n",
```

Comment:
Please remove spaces between keyword arguments and default parameters values, [per PE8](https://www.python.org/dev/peps/pep-0008/?#other-recommendations).

Reply:
Fixed